Cargo.toml

```diff
@@ -206,7 +206,7 @@ opendal = {version = "0.45", default-features = false, features = ["native-tls",
 
 # Tower
 tower = { version = "0.4.13", default-features = false, features = ["buffer", "limit", "retry", "timeout", "util", "balance", "discover"] }
-tower-http = { version = "0.4.4", default-features = false, features = ["decompression-gzip", "trace"]}
+tower-http = { version = "0.4.4", default-features = false, features = ["compression-gzip", "decompression-gzip", "trace"]}
```

Comment:
I think will be better to use `compression-full` feature instead. This should enable all supported by `tower-http` compression algorithms. In this case, a user will get automatically more supported compression algorithms.

More tests can be added in this case to test that the appropriate compression algorithm is chosen by `CompressionLayer` functionality. However, about tests I have no strict opinion on whether we need to test it on the Vector side or not - I'll leave it up to you ;)